knowledge-base/grid-custom-grouping-order.md

```diff
@@ -0,0 +1,188 @@
+					Width="200px" />
+		<GridColumn Title="String 3"
+					Field="@nameof(GridDataModel.S3)"
+					Width="200px" />
+		<GridColumn Title="String 4"
+					Field="@nameof(GridDataModel.S4)"
+					Width="200px" />
+		<GridColumn Title="String 5"
+					Field="@nameof(GridDataModel.S5)"
+					Width="200px" />
+	</GridColumns>
+</TelerikGrid>
+
+@code {
+	private ObservableCollection<GridDataModel> GridData = new ObservableCollection<GridDataModel>();
+
+	private bool UseGroupBy = true;
+
+	protected override void OnInitialized()
+	{
+		base.OnInitialized();
+
+		GridData = new ObservableCollection<GridDataModel>
+		{
+			new(5, "Hello world D", "Hello world 5-2", "Hello world 5-3", "Hello world 5-4", "Hello world 5-5"),
+			new(1, "Hello world A", "Hello world 1-2", "Hello world 1-3", "Hello world 1-4", "Hello world 1-5"),
+			new(3, "Hello world C", "Hello world 3-2", "Hello world 3-3", "Hello world 3-4", "Hello world 3-5"),
+			new(4, "Hello world C", "Hello world 4-2", "Hello world 4-3", "Hello world 4-4", "Hello world 4-5"),
+			new(2, "Hello world B", "Hello world 2-2", "Hello world 2-3", "Hello world 2-4", "Hello world 2-5"),
+		};
+	}
+
+	protected async Task ReadItems(GridReadEventArgs args)
+	{
+		var datasourceResult = GridData.ToDataSourceResult(args.Request);
+
+		// Determine if the data is grouped
+		if (args.Request.Groups.Any())
+		{
+			// Data is grouped, so we need to handle AggregateFunctionsGroup objects
+			var groups = datasourceResult.Data.Cast<AggregateFunctionsGroup>().ToList();
+
+			// Custom sort logic for grouped data based on your custom order: D, A, C, B
+			List<string> customOrder = new List<string> { "Hello world D", "Hello world A", "Hello world C", "Hello world B" };
+			groups = groups.OrderBy(group => customOrder.IndexOf(group.Key.ToString())).ToList();
+
+			args.Data = groups.Cast<object>().ToList();
+		}
+		else
+		{
+			// Data is not grouped, so we can cast directly to GridDataModel
+			var orderedData = datasourceResult.Data.Cast<GridDataModel>()
+								.OrderBy(item => GetCustomOrderIndex(item.S1))
+								.ToList();
+
+			args.Data = orderedData.Cast<object>().ToList();
+		}
+
+		args.Total = datasourceResult.Total;
+		args.AggregateResults = datasourceResult.AggregateResults;
+	}
+
+	private int GetCustomOrderIndex(string? value)
+	{
+		// Define the custom order
+		List<string> customOrder = new List<string> { "Hello world D", "Hello world A", "Hello world C", "Hello world B" };
+
+		// Return the index based on the custom order
+		return customOrder.IndexOf(value ?? string.Empty);
+	}
+
+	protected void OnGridStateInit(GridStateEventArgs<GridDataModel> args)
+	{
+		if (UseGroupBy)
+		{
+			GridState<GridDataModel> desiredState = new GridState<GridDataModel>()
+				{
+					GroupDescriptors = new List<GroupDescriptor>()
+				{
+					new GroupDescriptor()
+					{
+						Member = nameof(GridDataModel.S1),
+						MemberType = typeof(string),
+					}
+				}
+				};
```

Comment:
The tabs and brackets look off here. Is this the formatting that you get when you format the code in Visual Studio?